lib/aurora_bootstrapper/notifier.rb

```diff
@@ -12,7 +12,59 @@ def export_date
     
     # ENV is string to string dictionary
     def export_date_override
-      DateTime.now.strftime("%Y-%m-%d") if ENV.key?('EXPORT_DATE_OVERRIDE')
+      datetime_subfolder_str = nil
+
+      if ENV.key?('EXPORT_DATE_OVERRIDE')
+        now = Date.today
+        
+        # expiration time is 30 days
+        for i in 1..30
+          datetime_subfolder_str = check_db_dump_done_for_one_day( datetime: (now-i))
+          unless datetime_subfolder_str.nil?
+            break
+          end
+        end
+
+        # the first run for the given db partition
+        if datetime_subfolder_str.nil?
+          datetime_subfolder_str = now.strftime("%Y-%m-%d")
+        end
+      end
+
+      datetime_subfolder_str
+    end
+
+    def check_db_dump_done_for_one_day( datetime: )
+      datetime_subfolder_str = nil
+      prefix = [ bucket_path, datetime.strftime("%Y-%m-%d") ].join( '/' )
+
+      resp = client.list_objects_v2({
+        bucket: bucket,
+        prefix: prefix
+      })
+
+      if parse_resp(resp: resp)
+        datetime_subfolder_str = datetime.strftime("%Y-%m-%d")
+      end
+      
+      datetime_subfolder_str
+    end
+
+    def parse_resp( resp: )
```

Comment:
The name of the function suggests that it will be parsing a response; however it's actually telling us if the list of objects returned from a folder contains a key. Can we rename the function to more accurately reflect what it does, rather than how it does it?

Can we also consolidate the S3 responsibilities? As of right now we generate the prefix, and the object list in the parent, and then pass the result of the `list_objects_v2` call into this function. This generates a leaky abstraction where we expose to the caller that we are relying on S3. Ideally, the caller should never have to know how we arrive at the `done`ness for a date.
Can we pass the date into here and let it handle all the heavy lifting?

Reply:
Check the updated exists_db_dump_done_for_one_day?